schema/home/home_page_artist_module.js

```diff
@@ -4,43 +4,46 @@ import {
 } from 'lodash';
 import Artist from '../artist';
 import gravity from '../../lib/loaders/gravity';
+import { total } from '../../lib/loaders/total';
 import { NodeInterface } from '../object_identification';
 import { toGlobalId } from 'graphql-relay';
 import {
+  GraphQLEnumType,
   GraphQLID,
   GraphQLNonNull,
   GraphQLList,
   GraphQLObjectType,
   GraphQLString,
 } from 'graphql';
 
-export const Results = {
-  suggested: {
-    fetch: (accessToken, userID) => {
-      return gravity.with(accessToken)(`user/${userID}/suggested/similar/artists`);
-    },
+// This object is used for both the `key` argument enum and to do fetching.
+export const HomePageArtistModuleTypes = {
+  SUGGESTED: {
+    description: 'Artists recommended for the specific user.',
     display: (accessToken, userID) => {
-      return Promise.resolve(!!(accessToken && userID)).then(display => {
-        return display && Results.suggested.fetch(accessToken, userID).then(results => {
-          return results.length > 0;
-        });
-      });
+      if (!accessToken || !userID) {
+        return Promise.resolve(false);
+      }
+      return total(`user/${userID}/suggested/similar/artists?total_count=1&size=0`, accessToken)
```

Comment:
I don’t know why, but I thought Gravity response would be cached, even when authenticated, and so fetching the full list of artists (and then counting them) would have been more efficient than fetching just a count first and then the artists.

Clearly this assumption was wrong, so instead of downloading the list of artists twice, I now fetch just the count initially. Alas the amount of time to fulfil the Gravity request still takes the same amount of time ~500ms 😕